### PR TITLE
Feature/aapd 45 Function app for appeals FO and BO integration

### PR DESCRIPTION
**Pull request by chrishawldev**
The following PR contains the infra to achieve the following
1) New common app service plan for integrations functions apps, this is to host functions that listen to service bus topics to propagate data between front and back office applications, including but not limited to appeals. The idea being that all future function hosted integrations sit on tis plan so to not overwhelm the existing app service plan and to allow better separation of concerns so that the synchronous web service stacks and and asynchronous workloads that integrate with service bus topics can be hosted on there own infra, thus avoiding issues where for example a memory leak on one workload type would not then cause an outage on the other workload type because they no longer share compute resources. Furthermore, it is not uncommon that integrations with queue or topic technologies require different compute resources than those which serve traffic over a network protocol to front end clients.
2) New service bus topics for back office to front office integration.
3) New subnet for the new common functions integrations app service plan.

### CODE REVIEW

File: app/stacks/uk-west/appeals-service/variables.tf [hunk 1]
```diff
@@ -230,3 +235,29 @@ variable "clamav_host" {
...
+  type        = string
+}
+variable "appeals_feature_back_office_subscriber_enabled" {
+  description = "Feature toggle for appeals back office to front office service topic susbcription using azure functions"
+  type        = bool
+  default     = false
+}
+
+variable "common_integration_functions_subnet_id" {
+  description = "The id of the subnet for common integration functions app servcie plan"
```
Comment (fabfab-kc) [inline app/stacks/uk-west/appeals-service/variables.tf:261]: spelling...

File: app/components/appeals-app-services/variables.tf [hunk 2]
```diff
@@ -86,6 +91,11 @@ variable "integration_subnet_id" {
   type        = string
 }
 
+variable "common_integration_functions_subnet_id" {
+  description = "The id of the subnet for common integration functions app servcie plan"
```
Comment (fabfab-kc) [inline app/components/appeals-app-services/variables.tf:95]: spelling...

File: app/components/networking/outputs.tf [hunk 3]
```diff
@@ -3,6 +3,11 @@ output "back_office_integration_subnet_id" {
   value       = azurerm_subnet.back_office_integration_subnet.id
 }
 
+output "common_integration_functions_subnet_id" {
+  description = "Integration subnet for back office anti-virus resources"
```
Comment (fabfab-kc) [inline app/components/networking/outputs.tf:7]: Is this the right description?

**Review by fabfab-kc — APPROVED** [review]
LGTM! A couple of spelling mistakes in descriptions, and another little comment about the description of the app service plan

File: app/components/networking/outputs.tf (hunk 3 above)
Comment (chrishawldev) [reply app/components/networking/outputs.tf:7]: Well spotted

**Pull request merged**